### ISSUE TITLE
No compatible AppEngine Standard facet found

### ISSUE THREAD

**Issue by lmgagne** [GoogleCloudPlatform/google-cloud-eclipse#2285]
- Cloud Tools for Eclipse version: 1.3.0.201708031339
- Google Cloud SDK version: 165.0.0
- OS: MacOS 10.12.6
- Java version: 1.8.0_131-b11

We switched to gradle 3.1 to build our app.
It suddenly had different side effects as we had to reconfigure the project and convert it to a 'App engine standard project'.

Here how what we use to convert project:
- right-click on project
- click on properties
- click on configure
- click on 'convert to app engine standard project'

Below is the error message we get:
"Failed to convert project. No compatible AppEngine Standard facet found".

What are the cause of this error message?
How can we get back on track?

Thanks.

**Comment by chanseokoh** [comment]
I'd like to get more information.
- Does this happen when you freshly check out your project and run "Convert to App Engine Standard project"?
- Can you take a screenshot of (or describe) the current facet settings? (Right click on your project, _ Properties_ > _Project Facets_)
![selection_048](https://user-images.githubusercontent.com/10523105/29386905-bebb821e-82ab-11e7-9ada-9269a0936d4f.png)
- Try to install (check) the App Engine Java Standard Env facet together with the Dynamic Web Module facet (with versions matching your project). If you are blocked by some constraints, the dialog will give a hint why you can't.

**Comment by lmgagne** [comment]
This helped us in pinpointing what the issue is.

It happened that the Dynamic Web Module was downgraded for some unknown reasons. It was set to v2.4 while App Engine requires v2.5. What we did is to change this module to the required version (that is, v2.5).

Now, eclipse would not let me check the 'App Engine Java Standard Environment'. It still complains about Dynamic Web Module and Java versions even though it is properly set. Screenshot attached.

Thoughts?

<img width="816" alt="screen shot 2017-08-16 at 8 15 44 pm" src="https://user-images.githubusercontent.com/17264004/29390764-6bccb040-82c0-11e7-9964-5f0f3ce99bf4.png">

**Comment by chanseokoh** [comment]
@briandealwis probably have more insights.

**Comment by lmgagne** [comment]
Here is where I believe the issue is. 
File org.eclipse.wst.common.project.facet.core.xml was corrupted and had the following:
`<?xml version="1.0" encoding="UTF-8"?>`
`<faceted-project>`
`<fixed facet="jst.java"/>`
`<fixed facet="jst.web"/>`
`<installed facet="jst.java" version="1.7"/>`
`<installed facet="jst.web" version="2.5"/>`
`<installed facet="jst.java" version="1.8"/>`
`<installed facet="jst.web" version="3.0"/>`
`</faceted-project>`

Removing the last two lines fixed this issue.
We were able to go back to project facet and add App Engine.

Thanks for your help!

**Comment by chanseokoh** [comment]
Thanks for the update. Just FYI, if you switched to Gradle, CT4E will rely on WTP to build a deployable WAR (i.e., it won't do anything about Gradle at all).

**Comment by lmgagne** [comment]
Good point. Does it make sense then to use Gradle only as a build system (ie. facilitating the dependency management work) and leave all the deployment to CT4E/WTP?

**Comment by chanseokoh** [comment]
Actually, I don't know much about how Grade projects work with WTP when running them on a local server or deploying, so my previous answer could have beem wrong. Let me ask someone else.

In any case, I think how WTP publishes web modules can be seen in _Properties_ > _Deployment Assembly_, and AFAIK, that should be what CT4E/WTP uses for local run and deployment.

**Comment by briandealwis** [comment]
I don't know how Gradle/Buildship interact with WTP either.  But that mangled `org.eclipse.wst.common.project.facet.core.xml` should never happen.

**Comment by tidalgardener** [comment]
I've run into this issue now on my Mac w/ Java 8 SE

Eclipse IDE for Java Developers
Version: 2018-12 (4.10.0)
Build id: 20181214-0600 

Java SE 8 [1.8.0_202]

GWT SDK 2.8.1

Trying to convert to AppEngine standard:

Failed to convert project "mfxspaces".
Unable to install App Engine Standard facet
One or more constraints have not been satisfied.
App Engine Java Standard Environment JRE8 requires Java 1.8.

Any thoughts?

**Comment by briandealwis** [comment]
@tidalgardener could you confirm that your _Preferences > Java> Installed JREs_ and verify that your JRE is listed, and that the _Execution Environment_ page has it selected for the _JavaSE-1.8_ profile? And could you please post your `.settings/org.eclipse.wst.common.project.facet.core.xml`?

(And if you have a test case, that's also very useful.)

**Comment by tidalgardener** [comment]
@briandealwis Thank you for the follow up. Here is what I have on my Eclipse JRE. 
<img width="2560" alt="screen shot 2019-02-21 at 12 54 37" src="https://user-images.githubusercontent.com/11934276/53190640-02576200-35d8-11e9-8f26-735bb4b6fa9d.png">

The content of the file you asked for in that project is:

```
<?xml version="1.0" encoding="UTF-8"?>
<faceted-project>
  <installed facet="java" version="11"/>
  <installed facet="com.gwtplugins.gwt.facet" version="1.0"/>
</faceted-project>
```

AH! Look! It says Java version 11!! I don't even have that installed!!

**Comment by tidalgardener** [comment]
@briandealwis FYI So I did try changing it to 8, same issue also shows a red X on the project and doesn't say why.
<img width="2560" alt="screen shot 2019-02-21 at 13 09 43" src="https://user-images.githubusercontent.com/11934276/53191569-2c118880-35da-11e9-939f-e1c4d4b25e1a.png">
<img width="2560" alt="screen shot 2019-02-21 at 13 09 39" src="https://user-images.githubusercontent.com/11934276/53191570-2c118880-35da-11e9-9118-7d472d060006.png">

**Comment by chanseokoh** [comment]
```xml
<faceted-project>
  <installed facet="java" version="11"/>
```
From the XML, seems like your project is configured with Java 11 (i.e., has the Java 11 facet installed) at the beginning for some unknown reason. Probably your current facet settings (right click on your project, _Properties_ > _Project Facets_) will show Java 11. (Check this [screenshot](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/2285#issuecomment-322910687) where the Java facet is 1.7.)

**Comment by tidalgardener** [comment]
You are correct. 
<img width="2560" alt="screen shot 2019-02-21 at 14 22 40" src="https://user-images.githubusercontent.com/11934276/53195801-7435a880-35e4-11e9-88ff-8b192248a5a5.png">

I changed the value to 1.8 from 11, and then the conversion process worked. It is interesting, this was created as a test project on this machine which never had Java 11 on it.

Thank you again @chanseokoh and @briandealwis for your help.
